src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -328,6 +329,8 @@ public function request($method, $uri, array $parameters = array(), array $files
         }
 
         if ($this->followRedirects && $this->redirect) {
+            $this->redirects[serialize($this->history->current())] = true;
```

Comment:
did you consider `spl_object_hash` instead of serialize? It should be way faster and less memory intensive.

http://php.net/manual/en/function.spl-object-hash.php